src/adapters/github.ts

```diff
@@ -1,13 +1,14 @@
 /* eslint-disable class-methods-use-this */
-import Octokit from '@octokit/rest';
+/* eslint import/no-unresolved: [1, { ignore: ['dist-types/generated/method-types$'] }] */
```

Comment:
Was having issues with the `RestEndpointMethods` import after this, hence this line. Error:

```bash
    src/adapters/github.ts:3:37 - error TS2307: Cannot find module '@octokit/plugin-rest-endpoint-methods/dist-types/generated/method-types' or its corresponding type declarations.

    3 import { RestEndpointMethods } from '@octokit/plugin-rest-endpoint-methods/dist-types/generated/method-types';
``` 

Could use some help resolving this error.

Reply:
Taking a look.

Reply:
I think you want to `import { RestEndpointMethodTypes } from '@octokit/plugin-rest-endpoint-methods'` instead (see [1]).

References
1. https://github.com/octokit/plugin-rest-endpoint-methods.js/#typescript

Reply:
That's what I had initially but it was throwing all kinds of type errors.

Reply:
I see.  Looks like `RestEndpointMethodTypes['search']['repos']` is not compatible with `octokit.search.repo`.

Reply:
I tried to trackdown where the incompatibility stems from but wasn't successful.  Will try again later.

Reply:
E.g.
![image](https://user-images.githubusercontent.com/2030646/104130468-e0d40080-533e-11eb-922b-8e4a908116b4.png)

If you know how to fix please let me know. Updating the code was not too bad, it's these TS issues that I banged my head against for _far_ too long :cry:.

Reply:
I guess in the interim, you could do the following:

```suggestion
      let fullNamePath: string;
      let searchResults: string[];

      switch (search_type) {
        case 'repositories':
          searchResults = await this.octokit.paginate(
            this.octokit.search.repos,
            { q: search_query },
            (r: any) => {
              return r.data
            }
          )
          fullNamePath = 'full_name'
          break;
        case 'code':
        default:
          // github code search query. results are less reliable
          searchResults = await this.octokit.paginate(
            this.octokit.search.code,
            { q: search_query },
            (r: any) => {
              return r.data
            }
          )
          fullNamePath = 'repository.full_name'
      }
```

Reply:
Haha I see your point about this maybe not being a fix :smile:. I think per our conversation, I'll try to go the `any` route first as this feels a bit duplicative for no real gain. Hopefully I don't fall down another TS rabbit hole. Thanks for the deep dive into types!

Reply:
The `any` approach worked, resolving this.